syntax = "proto3";

service MembershipManagement {
    // Send heartbeat
    // A client to server function call
    rpc SendHeartBeat(HeartBeat) returns (ResultMsg) {}
}


message HeartBeat {
    string memoryUsage = 1;
    string cpuUsage = 2;
}

message ResultMsg {
    enum StatusCode {
        SUCCEEDED = 0;
        FAILED = 1;
    }
    StatusCode status = 1;
    string msg   = 2;
}